Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Confirm.hs

```diff
@@ -234,12 +245,37 @@ handler transporter req quote = do
             cs (showTimeIst booking.startTime) <> ".",
             "Check the app for more details."
           ]
-    buildRide driverId booking ghrId _ otp = do
+    buildRide driverId booking _ghrId _ otp = do
       guid <- Id <$> generateGUID
       shortId <- generateShortId
       -- let otp = T.takeEnd 4 customerPhoneNumber
       now <- getCurrentTime
+      endRideOtp <- generateOTPCode -- startRide otp code adding logic can be added later
       trackingUrl <- buildTrackingUrl guid
+      let (rideType, rideDetails) = case booking.bookingDetails of
+            BookingDetailsOnDemand {..} ->
+              ( DRide.ON_DEMAND,
+                DRide.RideDetailsOnDemand
+                  { toLocation = toLocation,
+                    driverGoHomeRequestId = Nothing,
+                    driverDeviatedFromRoute = Nothing,
+                    numberOfSnapToRoadCalls = Nothing,
+                    numberOfDeviation = Nothing,
+                    uiDistanceCalculationWithAccuracy = Nothing,
+                    uiDistanceCalculationWithoutAccuracy = Nothing
+                  }
+              )
+            BookingDetailsRental {} ->
```

Comment:
this will never happen, correct? because we do not create rental ride in `confirm` So we can create only  `BookingDetailsOnDemand` here